src/composed/shared.rs

```diff
@@ -86,6 +87,41 @@ pub trait Deserializable: Sized {
     fn from_packets<'a, I: Iterator<Item = Result<Packet>> + 'a>(
         packets: std::iter::Peekable<I>,
     ) -> Box<dyn Iterator<Item = Result<Self>> + 'a>;
+
+    /// Parses a single composition, from either ASCII-armored or binary OpenPGP data.
+    ///
+    /// Returns a composition and a BTreeMap containing armor headers
+    /// (None, if the data was unarmored)
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::type_complexity))]
+    fn from_reader_single<'a, R: Read + Seek + 'a>(
+        mut input: R,
+    ) -> Result<(Self, Option<BTreeMap<String, String>>)> {
```

Comment:
note for the future, we should really make a new type for the armor headers

Reply:
right, I opened #313 to track